# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/arcload/fix-ext2-inodesize.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- ./arcgrub/ext2.c.vanilla	2025-05-01 18:25:15.084516756 +0200
+++ ./arcgrub/ext2.c	2025-05-01 21:23:46.619956330 +0200
@@ -289,7 +289,7 @@
   if (grub_errno)
     return grub_errno;
 
-  inodes_per_block = EXT2_BLOCK_SIZE (data) / 128;
+  inodes_per_block = EXT2_BLOCK_SIZE (data) / grub_le_to_cpu16 (sblock->inode_size);
   blkno = (ino % grub_le_to_cpu32 (sblock->inodes_per_group))
     / inodes_per_block;
   blkoff = (ino % grub_le_to_cpu32 (sblock->inodes_per_group))
